### ISSUE TITLE
Interactive filter

### ISSUE THREAD

**Issue by Deleplace** [jhermsmeier/git-branch-select#7]
When list shows up alphabetically, it would be extra-super-nice to type 2 or 3 letters of desired branch, to filter the list and display only the branches that contain this exact substring in their name (case-insensitive).

```
? Select branch (Use arrow keys)
> Assemblyman
  Amusingness
  Antitruster
  Antielitist
  Autoclaving
```
pha
```
? Select branch (Use arrow keys)
> Elephant
  Pharmacy
```

**Comment by jhermsmeier** [comment]
Published as **git-branch-select@2.0.0**

**Comment by Deleplace** [comment]
Many thanks :)
